File: server/src/main/java/net/progruzovik/dissent/socket/Reader.java
package net.progruzovik.dissent.socket;

import net.progruzovik.dissent.captain.Player;
import org.springframework.lang.NonNull;

import java.util.Map;

@FunctionalInterface
public interface Reader {

    void read(@NonNull Player player, @NonNull Map<String, Integer> data);
}

File: server/src/main/kotlin/net/progruzovik/dissent/socket/reader/Reader.kt
package net.progruzovik.dissent.socket.reader

import net.progruzovik.dissent.captain.Player
import net.progruzovik.dissent.model.message.ClientSubject

interface Reader {

    val subject: ClientSubject

    fun read(player: Player, data: Map<String, Int>)
}
